Xlib/protocol/display.py

```
@@ -42,12 +42,41 @@
 from . import event
 
 if PY3:
-    def buffer(object, offset=None, size=None):
+
+    class bytesview(object):
+
+        def __init__(self, data, offset=None, size=None):
+            if offset is None:
+                offset = 0
+            if size is None:
+                size = len(data)-offset
+            if isinstance(data, bytes):
+                view = memoryview(data)
+            elif isinstance(data, bytesview):
+                view = data._view
+            else:
+                raise ValueError('unsupported type: %s' % type(data))
```

Maybe `raise TypeError('unsupported type: {}'.format(type(data)))` is better and should work on py2.6+.